server/log.go

```diff
@@ -76,8 +76,15 @@ func (s *Server) SetLogger(logger Logger, debugFlag, traceFlag bool) {
 	} else {
 		atomic.StoreInt32(&s.logging.trace, 0)
 	}
-
 	s.logging.Lock()
+	if s.logging.logger != nil {
+		// if we are the server's logger, call its Close function, but check
+		// the type because this could be a logger from another process
+		// embedding the NATS server, a SysLogger, or a dummy test logger.
+		if l, ok := s.logging.logger.(*srvlog.Logger); ok {
```

Comment:
if implementing `io.Closer` as mentioned above, you could just check whether it implements `io.Closer` interface and call `Close` here discarding the error,  that way whether `Close` is called or not would depend on  whether the passed logger satisfies the interface here (Syslogger seems to also implement: https://github.com/golang/go/blob/go1.9/src/log/syslog/syslog.go#L179-L189)

Reply:
Thank you - great suggestion.